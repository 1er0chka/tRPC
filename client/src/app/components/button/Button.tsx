"use client";
import React, { FunctionComponent, MouseEventHandler } from "react";
import styles from "./Button.module.scss";

interface IButtonProps {
  onClick: MouseEventHandler<HTMLButtonElement>;
  disabled: boolean;
  text: string;
}

const Button: FunctionComponent<IButtonProps> = ({
  onClick,
  disabled,
  text,
}) => {
  return (
    <button className={styles.button} onClick={onClick} disabled={disabled}>
      {text}
    </button>
  );
};

export default Button;
