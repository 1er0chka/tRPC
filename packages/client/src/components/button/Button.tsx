import React, {FunctionComponent, MouseEventHandler} from 'react';
import styles from './Button.module.scss'

interface IButtonProps {
    onClick: MouseEventHandler<HTMLButtonElement>;
    disabled?: boolean;
    label: string;
    mode: string;
    dataTestId?: string
}

const Button: FunctionComponent<IButtonProps> = ({onClick, label, disabled = false, mode, dataTestId = 'button'}) => {
    return (
        <button data-testid={dataTestId} className={mode === "primary" ? styles.primaryButton : styles.secondaryButton}
                onClick={onClick} disabled={disabled}>
            {label}
        </button>
    );
};

export default Button;