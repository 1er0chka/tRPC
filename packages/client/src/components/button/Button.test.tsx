import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Button from "./Button";

const onClick = jest.fn()

describe('Button Test', () => {
    test('Snapshot test', () => {
        render(
            <Button onClick={onClick} label={'Test'} mode={'primary'} dataTestId={'button'}/>
        )
        const button = screen.queryByTestId('button')
        expect(button).toMatchSnapshot()
    })
    test('Action test primary button', async () => {
        render(
            <Button onClick={onClick} label={'Test'} mode={'primary'} disabled={false}/>
        )
        const button = screen.queryByTestId('button')
        if (button) {
            await userEvent.click(button)
            expect(onClick).toHaveBeenCalled()
        }
    })
    test('Action test secondary button', async () => {
        render(
            <Button onClick={onClick} label={'Test'} mode={'secondary'}/>
        )
        const button = screen.queryByTestId('button')
        if (button) {
            await userEvent.click(button)
            expect(onClick).toHaveBeenCalled()
        }
    })
})
