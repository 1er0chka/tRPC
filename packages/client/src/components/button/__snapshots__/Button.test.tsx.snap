// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test Snapshot test 1`] = `
<button
  class="primaryButton"
  data-testid="button"
>
  Test
</button>
`;
